### ISSUE TITLE
[HOLD for payment 2024-03-22] [$500]  Chat - Green line marker does not go to the next message after deleting

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#31714]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.1-7
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Create a chat between User A and User B
2. As a user A, send message 1,2,3 to user B
3. Mark message 2 as unread, Notice the line appears above it
4. Delete message 2.
Notice the line disappears and does not go to message 3

## Expected Result:
The new line marker should go to message 3

## Actual Result:
The line disappears and does not go to message 3

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/32c65044-1555-4103-b835-262a5d474d24

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~014a86a25b91279911</li>
        <li>Upwork Job ID: 1727346941915725824</li>
        <li>Last Price Increase: 2023-11-29</li>
<li>Automatic offers: </li>
<ul>
<li>ntdiary | Reviewer | 27980016</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @MonilBhavsar</details>

**Comment by paultsimura** [comment]
Possible dupe of https://github.com/Expensify/App/issues/25606 – https://github.com/Expensify/App/issues/31637#issuecomment-1821396365

**Comment by gedu** [comment]
I will review it too

**Comment by gedu** [comment]
Seems that this bugs is caused by this commit: https://github.com/Expensify/App/commit/b4009201a2b848d9fe170de3677d8d6ebbf9e4cd, I will dig into how to improve it

**Comment by MonilBhavsar** [comment]
I assigned myself to issue. 

@gedu investigated this issue and he thinks previousAction is returned incorrectly after delete

> I was trying to use the previousReportActionID to handle the case when a message is deleted, because I can said assign to this message the unread state because your previous message was unread

> But after deleting the message, the previousReportActionID doesn’t get updated

**Comment by MonilBhavsar** [comment]
@gedu do you mind if I assign this issue to you

**Comment by MonilBhavsar** [comment]
@gedu is looking at this issue, so assigning. Had a chat

This is the investigation so far

> yes, let’s said I send this messages:
1 -> ID: a
2 -> ID: b
3 -> ID: c
4 -> ID: e
each message has that previousReportActionID, that means that each message points to the previous message id
1 -> previousReportActionID: -
2 -> previousReportActionID: a
3 -> previousReportActionID: b
4 -> previousReportActionID: c
If I delete the message 3, we end up like
(current)
1 -> previousReportActionID: -
2 -> previousReportActionID: a
4 -> previousReportActionID: c
But the message for should point to b  given that c doesn’t exist anymore
(expected)
1 -> previousReportActionID: -
2 -> previousReportActionID: a
4 -> previousReportActionID: b

previousReportActionID is incorrect when message is deleted. I have to look at it and fix it. I'll take a look today

**Comment by MonilBhavsar** [comment]
Can confirm this is issue https://github.com/Expensify/App/issues/31714#issuecomment-1842528916 and I am working on it

**Comment by slafortune** [comment]
@MonilBhavsar seems like we've covered a couple of GH's here, am I paying @gedu as the c+ only here?

**Comment by MonilBhavsar** [comment]
Working with internal project. Will circle back soon

**Comment by MonilBhavsar** [comment]
Working on it today

**Comment by MonilBhavsar** [comment]
Looking again. No progress because of holidays

**Comment by MonilBhavsar** [comment]
I'll circle back

**Comment by slafortune** [comment]
Is this something that should be set to Weekly @MonilBhavsar ?

**Comment by MonilBhavsar** [comment]
Let's keep it daily for now. I started with the query for this issue

**Comment by MonilBhavsar** [comment]
Dealing with waves, coming back

**Comment by MonilBhavsar** [comment]
Catching up from ooo, will work on this

**Comment by MonilBhavsar** [comment]
Finished my wave1 issue, I have now free cycles to use here

**Comment by MonilBhavsar** [comment]
Catching up. Will move here soon

**Comment by MonilBhavsar** [comment]
unblocking waves progress, will resume working

**Comment by MonilBhavsar** [comment]
> unblocking waves progress, will resume working

Same

**Comment by MonilBhavsar** [comment]
There was a recent discussion internally about `previousReportActionID`. I'll summarize here

**Comment by MonilBhavsar** [comment]
Me and eduardo had a call on Friday and we discussed the issue and we should be able to fix it completely on frontend side only.

**Comment by gedu** [comment]
I will be working on this 💪

**Comment by gedu** [comment]
Tomorrow I will create the PR

**Comment by gedu** [comment]
@MonilBhavsar PR updated

**Comment by MonilBhavsar** [comment]
Thanks! I'll take a look soon
Also cc @mananjadhav

**Comment by MonilBhavsar** [comment]
@gedu can you please link the PR to this issue

**Comment by gedu** [comment]
https://github.com/Expensify/App/pull/25771

**Comment by mananjadhav** [comment]
I am confused if the PR is on hold or not?

**Comment by MonilBhavsar** [comment]
It's not on HOLD. took off hold from title

**Comment by MonilBhavsar** [comment]
PR is merged!

**Comment by slafortune** [comment]
@mananjadhav Can you complete this checklist so I can confirm payment today?

**Comment by mananjadhav** [comment]
@slafortune I'll let @MonilBhavsar confirm if we need a regression test here. This is a part of the project for `Unread Markers`, and the @MonilBhavsar @gedu have been working on this with the previous PRs as well.

**Comment by MonilBhavsar** [comment]
Regression tests are being added here https://github.com/Expensify/Expensify/issues/378297

**Comment by mananjadhav** [comment]
@slafortune No further action needed here. Can you please add the payout summary?

**Comment by slafortune** [comment]
Offer sent @mananjadhav https://www.upwork.com/nx/wm/offer/101523017

**Comment by mananjadhav** [comment]
@slafortune I'll be requesting payout on ND. Hence, all I need is the payout summary. You can rescind the Upwork offer.

**Comment by slafortune** [comment]
Shoot - I knew that!

**Comment by slafortune** [comment]
@gedu does not require payment (Contractor)
@mananjadhav requires payment through NewDot Manual Requests - $500

**Comment by JmillsExpensify** [comment]
$500 approved for @mananjadhav based on summary.
